### ISSUE TITLE
RenderChunk.field_188284_q: needsUpdateCustom -> needsImmediateUpdate (+ 2 more associated)

### ISSUE THREAD

**Issue by williewillus** [ModCoderPack/MCPBot-Issues#333]
nothing is "custom" about this boolean. There is a normal `needsUpdate` field, and there is this field. 

How is this field used?
* **TLDR If it is true, then `RenderGlobal` will re-render the affected chunks in the main thread, instead of offloading them onto the worker pool.**

* This field is a direct result of setting flag 8 (bit 3) in notifyBlockUpdate/setBlockState. You can see this by tracing from `RenderGlobal.notifyBlockUpdate` -> `ViewFrustum.markBlocksForUpdate` -> `RenderChunk.setNeedsUpdate`. Several vanilla actions call `setBlockState` with this flag present, when immediate rerendering is desired. For example: player placing a block, player opening a door or fence gate, etc.
* `RenderGlobal.setupTerrain` loops through all `RenderChunk` and, on the main thread, rebuilds all chunks 768 or closer to the player with both `needsUpdate` and `needsUpdateCustom` set. All other chunks that need updates, but either aren't close enough or don't have `needsUpdateCustom` set, are added to a set.
* `RenderGlobal.updateChunks` then loops through this set. If the `RenderChunk` has `needsUpdateCustom` set, it immediately rebuilds the chunk on the main thread. Otherwise, a task is enqueued to be submitted to the Chunk Batcher threads later.

Therefore, I request the following renames:
* `RenderChunk.func_178575_a.p_178575_1_`: `setNeedsUpdate(needsUpdateIn)` -> `setNeedsUpdate(updateImmediately)`.
* `RenderChunk.field_188284_q`: `needsUpdateCustom` -> `needsImmediateUpdate`
* `RenderChunk.func_188281_o`: `isNeedsUpdateCustom()` -> `needsImmediateUpdate()` (if possible to have same name as a field, otherwise `isNeedsImmediateUpdate()`)

**Comment by williewillus** [comment]
Also, if methods and fields can share name:

`RenderChunk.func_178569_m`: `isNeedsUpdate()` -> `needsUpdate()`

because isNeedsUpdate sounds horrible

**Comment by kashike** [comment]
done.
- [x] `field_188284_q`: `needsUpdateCustom` -> `needsImmediateUpdate`
- [x] `func_188281_o`: `isNeedsUpdateCustom` -> `needsImmediateUpdate`
- [x] `func_178569_m`: `isNeedsUpdate` -> `needsUpdate`
